tests/src/drivers/aead.c

```diff
@@ -17,6 +17,8 @@
  *  limitations under the License.
  */
 
+#define MBEDTLS_ALLOW_PRIVATE_ACCESS
```

Comment:
I was not sure where to put this.
tests source files don't have a common.h like the sources in the library

Reply:
They do: `tests/include/helpers.h` and a few others. We have to define it before including any of `mbedtls/*.h` other than `config.h`. We can even put the `#define` at the top of `tests/suite/helpers.function`, this way it won't be sensitive to the order in which header files are included. Except that I see that `main_test.function` includes `psa_crypto.h`, but I think it's obsolete and can just be removed (and if not it should move into `helpers.function` or `psa_crypto_helpers.h`).

Ideally we'd have some test suites that are invasive and have `MBEDTLS_ALLOW_PRIVATE_ACCESS` defined, and some that do things solely with public APIs and don't have it defined. It's not necessary to do this work now, though, so for now let's just define `MBEDTLS_ALLOW_PRIVATE_ACCESS` in a global header. I've filed this as https://github.com/ARMmbed/mbedtls/issues/4435.

Reply:
Ah, that psa_crypto include was exactly the reason why I placed it in each of .c files.
Alright, I'll remove/move psa_crypto.h from main_test and see how it goes. thanks!